bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/transport/PanamaxFurmanSocket.java

```diff
@@ -0,0 +1,203 @@
+
+/**
+ * Wraps the communication to a Panamax/Furman Power Conditioner with a BlueBOLT-CV1 or BlueBOLT-CV2 card via telnet
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public class PanamaxFurmanSocket {
+    private static final int CONNECT_TIMEOUT_MILLIS = (int) TimeUnit.SECONDS.toMillis(10);
+    private static final int INFINITE_TIMEOUT = 0;
+
+    protected final Object ioMutex = new Object();
+    private final Logger logger = LoggerFactory.getLogger(PanamaxFurmanSocket.class);
+
+    private final InputStreamReaderThread inputStreamReaderThread;
+    private final BufferedWriter writer;
+    private final PanamaxFurmanCommunicationEventListener eventListener;
+
+    private final String connectionName;
+
+    private Socket telnetSocket;
+
+    /**
+     * Creates a new connection to the power conditioner device
+     */
+    public PanamaxFurmanSocket(String connectionName, String address, int port,
+            PanamaxFurmanCommunicationEventListener eventListener) throws IOException, InterruptedException {
+        this.eventListener = eventListener;
+        this.connectionName = connectionName;
+        synchronized (ioMutex) {
+            logger.trace("Attempting to connect via telnet to {}:{}  @{}", address, port, connectionName);
+            try {
+                telnetSocket = new Socket();
+                telnetSocket.connect(new InetSocketAddress(address, port), CONNECT_TIMEOUT_MILLIS);
+                telnetSocket.setKeepAlive(true);
+                // Block forever on reads
+                telnetSocket.setSoTimeout(INFINITE_TIMEOUT);
+                InputStream is = telnetSocket.getInputStream();
+                OutputStream os = telnetSocket.getOutputStream();
+                this.inputStreamReaderThread = new InputStreamReaderThread(is);
+                this.inputStreamReaderThread.start();
+                logger.trace("Started reader thread, pausing   @{}", connectionName);
+                Thread.sleep(500);
+                logger.trace("Pause complete   @{}", connectionName);
+                this.writer = new BufferedWriter(new OutputStreamWriter(os, StandardCharsets.US_ASCII));
+                logger.debug("Connected   @{}", connectionName);
+            } catch (IOException | InterruptedException e) {
+                // should any errors occur, clean everything up
+                close();
+                throw e;
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    protected void send(@Nullable String textToSend) throws IOException {
+        synchronized (ioMutex) {
+            String command = textToSend;
+            try {
+                writer.write(command + "\r");
+                writer.flush();
+                logger.debug("sent: {}    @{}", command, connectionName);
+            } catch (IOException e) {
+                close();
+                throw e;
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
```

Comment:
Pleaswe remove null surpression and properly handle null warning by creating a local variabel